src/sentry/search/events/datasets/metrics.py

```diff
@@ -28,7 +28,8 @@ def search_filter_converter(
             constants.PROJECT_NAME_ALIAS: self._project_slug_filter_converter,
             constants.EVENT_TYPE_ALIAS: self._event_type_converter,
             constants.TEAM_KEY_TRANSACTION_ALIAS: self._key_transaction_filter_converter,
-            "transaction.duration": self._duration_filter_converter,
+            "transaction.duration": self._duration_filter_converter,  # Only for dry_run
+            "environment": self.builder._environment_filter_converter,
```

Comment:
Do we need to explicitly map this here? If you've overridden it in the MQB, won't it call the overridden method instead of the one in the parent class?

I removed it locally and the tests still pass

Reply:
Good catch, so that's not 100% correct. The environment converter is on the builder because we need it when resolving params (ie. using page filters to set environment). While the search_filter_converter is for filters in the query string (ie. sending "environment:staging")

But you pointing out this still passed helped me notice a bug with the null case I had to update 🙏